### PR TITLE
[docs][WIP] Update existing documentation

### PR DESCRIPTION
**Pull request by ynotdraw**
What are your thoughts on moving the docs out of the `docs` directory and creating a root-level `API.md` file instead to document all-the-things? I wonder if putting them in separate files may lead to hopping around a bit too much between tabs?  The separation could be good though.  I keep going back and forth.

Also, what all do you want documented before this branch is merged into master?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,24 +32,63 @@ module.exports = function(environment) {
...
-        This is where you are!
-      {{/marker.popup}}
-    {{/map.marker}}
+```javascript
+'mapbox-gl': {
+  accessToken: 'ACCESS TOKEN HERE',
+  map: {
+    style: 'mapbox://styles/mapbox/basic-v9',
+    zoom: 13,
+    center: [ -96.7969879, 32.7766642 ]
```
Comment (ynotdraw) [inline README.md:50]: Added these for a complete working example (map options, a controller with a point, and then a layer for a blue dot on the point).

**Comment by kturney** [comment]
I guess a single API.md may be fine.

**Comment by ynotdraw** [comment]
Single `API.md` file [rendered](https://github.com/ynotdraw/ember-mapbox-gl/blob/10968b577ea215397ba778a12446af112b8ace3d/API.md) for your review. I still need to add docs for `control`, `call`, and `marker`. Maybe before I get too far, I'll wait for feedback on what we have now.

Tried creating full examples and linking to documentation for additional options. Maybe we want to actually document more complex options, such as `source.layer` having many types? Or think providing the docs link is enough? FWIW, in the case of an image, a consumer would see how set a layer type to `symbol` and some additional properties passed down.


**TODO:**

- [ ] control docs
- [ ] call docs
- [ ] marker docs

**Comment by kturney** [comment]
Can you rebase on master instead of merging?

**Review by kturney — APPROVED** [review]

**Pull request merged**